tests/components/cloud/test_backup.py

```diff
@@ -0,0 +1,228 @@
+"""Test the cloud backup platform."""
+
+from collections.abc import AsyncGenerator, Generator
+from typing import Any
+from unittest.mock import Mock, PropertyMock, patch
+
+import pytest
+
+from homeassistant.components.backup import DOMAIN as BACKUP_DOMAIN
+from homeassistant.components.cloud import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
+
+from tests.conftest import AiohttpClientMocker
+from tests.typing import MagicMock, WebSocketGenerator
+
+
+@pytest.fixture(autouse=True)
+async def setup_integration(
+    hass: HomeAssistant, cloud: MagicMock
+) -> AsyncGenerator[None]:
+    """Set up cloud integration."""
+    with patch("homeassistant.components.backup.is_hassio", return_value=False):
+        assert await async_setup_component(hass, BACKUP_DOMAIN, {BACKUP_DOMAIN: {}})
+        assert await async_setup_component(hass, DOMAIN, {DOMAIN: {}})
+        await hass.async_block_till_done()
+        yield
+
+
+async def test_agents_info(
+    hass: HomeAssistant,
+    hass_ws_client: WebSocketGenerator,
+) -> None:
+    """Test backup agent info."""
+    client = await hass_ws_client(hass)
+
+    await client.send_json_auto_id({"type": "backup/agents/info"})
+    response = await client.receive_json()
+
+    assert response["success"]
+    assert response["result"] == {
+        "agents": [{"agent_id": "backup.local"}, {"agent_id": "cloud.cloud"}],
+        "syncing": False,
+    }
+
+
+async def test_agents_list_backups(
+    hass: HomeAssistant,
+    hass_ws_client: WebSocketGenerator,
+    cloud: MagicMock,
+) -> None:
+    """Test agent list backups."""
+    client = await hass_ws_client(hass)
+
+    with patch(
+        "homeassistant.components.cloud.backup.async_files_list", spec_set=True
```

Comment:
We can use the `mock_list_files` fixture and just change the return value if needed.